### PR TITLE
New blog post: Upcoming Features in Go 1.18

### PR DESCRIPTION
**Pull request by SebastianM**
# Publish Checkliste

- **Meta** (abzuhaken durch den Autor)
  - [x] [Artikeldatei am richtigen Ort abgelegt](https://github.com/adessoAG/devblog/blob/master/examples/2017-08-10-blog-post-guide.md#dateiname-und-ablageort)
  - [x] [Autoreninfo gepflegt](https://github.com/adessoAG/devblog/blob/master/examples/2017-08-10-blog-post-guide.md#autoren-informationen)
  - [x] [Metadaten gepflegt:](https://github.com/adessoAG/devblog/blob/master/examples/2017-08-10-blog-post-guide.md#metadaten)
    - [x] "title" vergeben
    - [x] "author_ids" angegeben
    - [x] "categories" enthält maximal eine der Hauptkategorien (Architektur, Branchen, Digital Experience, Inside adesso, KI, Methodik, Softwareentwicklung)
    - [x] "tags" enthält (beliebige) passende Tags
  - [] Der blogpost-checker hat die Richtigkeit der Metadaten bestätigt (siehe CI Check)
- **Text** (abzuhaken durch den Autor)
  - [x] [Teaser entspricht den Vorgaben](https://github.com/adessoAG/devblog/blob/master/examples/2017-08-10-blog-post-guide.md#einleitung--teaser)
  - [x] Rechtschreibung korrigiert
  - [x] [Ein Satz pro Zeile](https://github.com/adessoAG/devblog/blob/master/examples/2017-08-10-blog-post-guide.md#ein-satz-pro-zeile)
  - [x] [Bilder sind korrekt in den Artikel eingebunden](https://github.com/adessoAG/devblog/blob/master/examples/2017-08-10-blog-post-guide.md#bilder)
  - [x] [Überschriften entsprechen den Vorgaben](https://github.com/adessoAG/devblog/blob/master/examples/2017-08-10-blog-post-guide.md#%C3%9Cberschriften)
  - [x] Code-Snippets wurden mit [Syntax Highlighting](https://github.com/adessoAG/devblog/blob/master/examples/2017-08-10-blog-post-guide.md#syntax-highlighting) versehen
  - [x] Der Artikel wurde in der [Vorschau](https://github.com/adessoAG/devblog/blob/master/examples/2017-08-10-blog-post-guide.md#online-preview) geprüft 
  - [x] Auf gendergerechte Formulierungen wurde geachtet
- **Review** (abzuhaken durch den Reviewer)
  - [ ] Rechtschreibung korrigiert
  - [ ] Der Artikel wurde in der [Vorschau](https://github.com/adessoAG/devblog/blob/master/examples/2017-08-10-blog-post-guide.md#online-preview) geprüft 
  - [ ] Auf gendergerechte Formulierungen wurde geachtet
  - [x] Der Autor ist bei CCO hinterlegt
  - [ ] CCO wurde über neuen Artikel informiert
  - [ ] Date-Angaben 
    - [ ] stimmen im Header und im Dateinamen überein
    - [ ] stehen auf dem heutigen Tag (aber in der Vergangenheit)

### CODE REVIEW

File: _posts/2021-11-21-upcoming-features-in-go-1-18.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+Go 1.18 will be a significant release of the programming language that will contain some major features that I'm excited about. 
+The upcoming version is scheduled for early 2022. 
+The first beta should be out in a month. 
+Let's take a look at some of the new features that will be available.
+
+## Generics
+
+The long-awaited generics support will land in Go 1.18. 
+The lack of generics in Go was the biggest point of criticism of the developer community. 
+It took some years from the design phase to the actual implementation that will land in Go 1.18.
```
Comment (jo2) [inline _posts/2021-11-21-upcoming-features-in-go-1-18.md:20]: will land in Go 1.18. hattest du zwei Sätze vorher schon, kannst du das bitte umformulieren?

File: _posts/2021-11-21-upcoming-features-in-go-1-18.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+The topic is too large to explain in detail in this blog post. 
+There are already a lot of good blog posts about it. 
+The following is my favorite that covers all relevant aspects of it: https://bitfieldconsulting.com/golang/generics.
```
Comment (jo2) [inline _posts/2021-11-21-upcoming-features-in-go-1-18.md:24]:
```suggestion
The following is my favorite that covers all relevant aspects of it: [https://bitfieldconsulting.com/golang/generics](https://bitfieldconsulting.com/golang/generics). 
```

File: _posts/2021-11-21-upcoming-features-in-go-1-18.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+## Official Fuzzing Support
+
+Official fuzzing support will also be available in Go 1.18. 
+The fuzzing features will be considered experimental, and the API will not be covered by the Go 1 compatibility promise yet. 
+It should serve as a proof-of-concept and the Go team asks for feedback from the community.
+
+If you haven't heard of fuzzing yet, the [blog post](https://go.dev/blog/fuzz-beta) of the beta announcement describes it very well:
+
+> Fuzzing is a type of automated testing which continuously manipulates inputs to a program to find issues such as panics or bugs. These semi-random data mutations can discover new code coverage that existing unit tests may miss, and uncover edge case bugs which would otherwise go unnoticed. Since fuzzing can reach these edge cases, fuzz testing is particularly valuable for finding security exploits and vulnerabilities.
```
Comment (jo2) [inline _posts/2021-11-21-upcoming-features-in-go-1-18.md:66]: Blockquotes sind im adesso-Blog leider nicht implementiert. Du kannst beispielsweise einen Kursivblock verwenden, um das Zitat zu kennzeichnen.

File: _posts/2021-11-21-upcoming-features-in-go-1-18.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+You can read the design doc by Katie Hockman [here](https://go.googlesource.com/proposal/+/master/design/draft-fuzzing.md). There's also [Go Time podcast episode](https://changelog.com/gotime/187) with Katie that covers this topic.
```
Comment (jo2) [inline _posts/2021-11-21-upcoming-features-in-go-1-18.md:68]:
```suggestion
You can read the design doc by Katie Hockman [here](https://go.googlesource.com/proposal/+/master/design/draft-fuzzing.md). There's also a [Go Time podcast episode](https://changelog.com/gotime/187) with Katie that covers this topic.
```

File: _posts/2021-11-21-upcoming-features-in-go-1-18.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+```bash
+time go fmt ./pkg/...
+```
+
+With Go 1.17 it took **56 seconds** to format all files. 
+With the latest `gotip` version, it took **1 minute and 20 seconds**. 
+I also had to increase the ulimit on my machine to prevent a crash. Let's see how this feature evolves until the stable release.
+
+## Try out the Upcoming Features
```
Comment (jo2) [inline _posts/2021-11-21-upcoming-features-in-go-1-18.md:92]: Du erwähnst im vorherigen Absatz bereits gotip, kannst du diesen Absatz zuerst schreiben, damit dann im Absatz zu Faster Runs direkt klar ist, was das ist?

**Review by jo2 — CHANGES_REQUESTED** [review]
Ein paar Kleinigkeiten habe ich noch angemerkt, ansonsten liest sich der Artikel recht gut.

**Comment by SebastianM** [comment]
@jo2  done

**Review by jo2 — DISMISSED** [review]

**Comment by jo2** [comment]
@acetous guckst du hier noch drüber?

**Review by acetous — APPROVED** [review]

**Review by acetous — APPROVED** [review]
